File: Install_Android/app/src/main/java/com/goldenplanet/www/gp_android/PackageReceiver.java
package com.goldenplanet.www.gp_android;

import android.app.Activity;
import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.util.Log;

public class PackageReceiver extends BroadcastReceiver {

    private static String TAG = "PackageReceiver";
    @Override
    public void onReceive(Context context, Intent intent) {
        String packageName = intent.getData().getSchemeSpecificPart();
        String action = intent.getAction();
        if(action.equals(Intent.ACTION_PACKAGE_ADDED)) {
            Log.d("GADATA", "앱설치");
        }
        else if(action.equals(Intent.ACTION_PACKAGE_REMOVED)) {
            Log.d("GADATA", "앱삭제");

            SharedPreferences prefs = context.getSharedPreferences("isFirst", Context.MODE_PRIVATE);
            boolean isFirst = prefs.getBoolean("isFirst", false);
            if(isFirst){
                prefs.edit().remove("isFirst").commit();
            }
        }

    }
}

File: Install_AOSkt/app/src/main/java/com/goldenplanet/www/gp_android/PackageReceiver.kt
package com.goldenplanet.Install_AOSkt

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.util.Log

class PackageReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        val packageName = intent.data!!.schemeSpecificPart
        val action = intent.action
        if (action == Intent.ACTION_PACKAGE_ADDED) {
            Log.d("GADATA", "앱설치")
        } else if (action == Intent.ACTION_PACKAGE_REMOVED) {
            Log.d("GADATA", "앱삭제")
            val prefs = context.getSharedPreferences("isFirst", Context.MODE_PRIVATE)
            val isFirst = prefs.getBoolean("isFirst", false)
            if (isFirst) {
                prefs.edit().remove("isFirst").commit()
            }
        }
    }

    companion object {
        private const val TAG = "PackageReceiver"
    }
}